src/main.rs

```diff
@@ -240,20 +239,49 @@ async fn run_server(addr: SocketAddr) -> anyhow::Result<()> {
         .await
         .context("database migrations")?;
 
+    // spawning a background task that will schedule the jobs
+    // every JOB_SCHEDULING_CADENCE_IN_SECS
+    task::spawn(async move {
+        loop {
```

Comment:
This "loop with retry" is almost exactly the same as the one for processing the jobs, changing only the cadence and the "callback" that is called (schedule_jobs or run_scheduled_jobs).

We wanted to abstract this and create a `loop_with_retry` function that receives a cadence and a callback function to avoid the duplication. Do you think it's a good idea? We haven't managed to get it working clear for now, but we'll keep trying if it makes sense

Reply:
I wouldn't worry about it.